src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1092,19 +1110,101 @@ private int getMostEfficientTool(@NotNull final Block target)
         final int required = target.getHarvestLevel(target.getDefaultState());
         int bestSlot = -1;
         int bestLevel = Integer.MAX_VALUE;
-        @NotNull final InventoryCitizen inventory = worker.getInventoryCitizen();
+        @NotNull InventoryCitizen inventory = worker.getInventoryCitizen();
+
         for (int i = 0; i < inventory.getSizeInventory(); i++)
         {
             final ItemStack item = inventory.getStackInSlot(i);
             final int level = Utils.getMiningLevel(item, tool);
+
             if (level >= required && level < bestLevel)
             {
-                bestSlot = i;
-                bestLevel = level;
+                boolean isUsable = true;
+
+                if (item != null)
+                {
+                    isUsable = verifyToolLevel(item, level);
+                }
+                if (isUsable)
```

Comment:
Okay I got it now! You need to pull the nullcheck into verifyToolLevel. It should definitely be @Nullable and return true when the item is null